os/arch/arm/src/imxrt/imxrt_mtd_flash.c

```diff
@@ -0,0 +1,688 @@
+		}
+	}
+
+	return nbytes;
+}
+#endif							/* CONFIG_MTD_BYTE_WRITE */
+
+/************************************************************************************
+ * Name: imxrt_ioctl
+ ************************************************************************************/
+
+static int imxrt_ioctl(FAR struct mtd_dev_s *dev, int cmd, unsigned long arg)
+{
+	FAR struct imxrt_dev_s *priv = (FAR struct imxrt_dev_s *)dev;
+	int ret = -EINVAL;			/* Assume good command with bad parameters */
+	char tempLog[128];
+
+	fvdbg("cmd: %d \n", cmd);
+
+	switch (cmd) {
+	case MTDIOC_GEOMETRY: {
+		FAR struct mtd_geometry_s *geo = (FAR struct mtd_geometry_s *)((uintptr_t)arg);
+		if (geo) {
+			/* Populate the geometry structure with information need to know
+			 * the capacity and how to access the device.
+			 *
+			 * NOTE: that the device is treated as though it where just an array
+			 * of fixed size blocks.  That is most likely not true, but the client
+			 * will expect the device logic to do whatever is necessary to make it
+			 * appear so.
+			 */
+
+			geo->blocksize = (1 << priv->pageshift);
+#ifdef CONFIG_IMXRT_SUBSECTOR_ERASE
+			if (priv->subsectorshift > 0) {
+				geo->erasesize = (1 << priv->subsectorshift);
+				geo->neraseblocks = priv->nsectors * (1 << (priv->sectorshift - priv->subsectorshift));
+			} else
+#endif
+			{
+				geo->erasesize = (1 << priv->sectorshift);
+				geo->neraseblocks = priv->nsectors;
+			}
+
+			ret = OK;
+
+			snprintf(tempLog, sizeof(tempLog), "sectorshift: %d nsectors: %d pageshift: %d, npages: %d\n", priv->sectorshift, priv->nsectors, priv->pageshift, priv->npages);
+			IMXLOG(tempLog);
+
+			snprintf(tempLog, sizeof(tempLog), "blocksize: %d erasesize: %d neraseblocks: %d\n", geo->blocksize, geo->erasesize, geo->neraseblocks);
+			IMXLOG(tempLog);
+			fvdbg("blocksize: %d erasesize: %d neraseblocks: %d\n", geo->blocksize, geo->erasesize, geo->neraseblocks);
+		}
+	}
+	break;
+
+	case MTDIOC_BULKERASE: {
+		/* Erase the entire device */
+		ret = imxrt_bulkerase(priv);
+	}
+	break;
+
+	case MTDIOC_XIPBASE:
+	default:
+		ret = -ENOTTY;			/* Bad command */
+		break;
+	}
+
+	fvdbg("return %d\n", ret);
+	return ret;
+}
+
+/************************************************************************************
+ * Public Functions
+ ************************************************************************************/
+
+/************************************************************************************
+ * Name: imxrt_initialize
+ *
+ * Description:
+ *   Create an initialize MTD device instance.  MTD devices are not registered
+ *   in the file system, but are created as instances that can be bound to
+ *   other functions (such as a block or character driver front end).
+ *
+ ************************************************************************************/
+
+FAR struct mtd_dev_s *imxrt_mtd_initialize(void)
+{
+	FAR struct imxrt_dev_s *priv;
+	int ret;
+
+	fvdbg("dev: %p\n", dev);
```

Comment:
"dev" is not defined here. 
It causes compile error.

Reply:
fixed it